src/admin/permission-groups/views/PermissionGroupEdit.tsx

```diff
@@ -0,0 +1,426 @@
+import {
+	Permission,
+	PermissionGroup,
+	PermissionGroupEditFormErrorState,
+	PermissionsTableCols,
+} from '../permission-group.types';
+
+interface PermissionGroupEditProps extends DefaultSecureRouteProps<{ id: string }> {}
+
+const PermissionGroupEdit: FunctionComponent<PermissionGroupEditProps> = ({
+	history,
+	match,
+	location,
+}) => {
+	const [t] = useTranslation();
+
+	// Hooks
+	const [initialPermissionGroup, setInitialPermissionGroup] = useState<PermissionGroup | null>(
+		null
+	);
+	const [permissionGroup, setPermissionGroup] = useState<PermissionGroup | null>(null);
+
+	const [formErrors, setFormErrors] = useState<PermissionGroupEditFormErrorState>({});
+	const [isSaving, setIsSaving] = useState<boolean>(false);
+	const [allPermissions, setAllPermissions] = useState<Permission[]>([]);
+	const [selectedPermissionId, setSelectedPermissionId] = useState<string | null>(null);
+	const [loadingInfo, setLoadingInfo] = useState<LoadingInfo>({ state: 'loading' });
+	const [sortColumn, sortOrder, handleSortClick] = useTableSort<PermissionsTableCols>('label');
+
+	const isCreatePage: boolean = location.pathname.includes(`/${ROUTE_PARTS.create}`);
+
+	const initOrFetchPermissionGroup = useCallback(async () => {
+		if (isCreatePage) {
+			const permGroup = ({
+				label: '',
+				description: '',
+				created_at: new Date().toISOString(),
+				updated_at: new Date().toISOString(),
+				permissions: [],
+			} as unknown) as PermissionGroup;
+			setInitialPermissionGroup(permGroup);
+			setPermissionGroup(permGroup);
+		} else {
+			try {
+				const response = await dataService.query({
+					query: GET_PERMISSION_GROUP_BY_ID,
+					variables: { id: match.params.id },
+				});
+
+				const permissionGroupObj = get(response, 'data.users_permission_groups[0]');
+
+				if (!permissionGroupObj) {
+					setLoadingInfo({
+						state: 'error',
+						icon: 'search',
+						message: t('Deze permissie groep werd niet gevonden'),
+					});
+					return;
+				}
+
+				const permissions: Permission[] = flatten(
+					get(permissionGroupObj, 'permission_group_user_permissions', []).map(
+						(permissionGroupItem: any) => {
+							return get(permissionGroupItem, 'permissions', []);
+						}
+					)
+				);
+
+				const permGroup = {
+					permissions,
+					id: permissionGroupObj.id,
+					label: permissionGroupObj.label,
+					description: permissionGroupObj.description,
+					created_at: permissionGroupObj.created_at,
+					updated_at: permissionGroupObj.updated_at,
+				};
+				setInitialPermissionGroup(permGroup);
+				setPermissionGroup(permGroup);
+			} catch (err) {
+				console.error(
+					new CustomError('Failed to get permission group by id', err, {
+						query: 'GET_PERMISSION_GROUP_BY_ID',
+						variables: { id: match.params.id },
+					})
+				);
+				setLoadingInfo({
+					state: 'error',
+					message: t('Het ophalen van de permissie groep is mislukt'),
+				});
+			}
+		}
+	}, [setLoadingInfo, setPermissionGroup, t]);
```

Comment:
`React Hook useCallback has missing dependencies: 'isCreatePage' and 'match.params.id'`